### DISCUSSION TITLE
Decrypting data encrypted with the user's PubKey in Snaps

### DISCUSSION THREAD

**Discussion by shawnmitchell** [MetaMask/snaps#313] [Q&A] [answered]
Can anyone point me in the right direction to do data decryption in a Snap with the user's private key?

I've tried importing @metamask/eth-sig-util but SES doesn't like that.

I've tried wallet.request({ method: 'eth_decrypt' ...}); but that's not it either.

I even tried passing the window.ethereum object as a param to the Snap call 😂 but again, no dice.

This seems like a natural use case for Snaps and I hope I'm just being stupid and not seeing the obvious answer.

I am using the ethers-js example Snap as my starting point, and it's very instructive. Thanks to the author!

My Snap will take an encrypted array of user data that was used to generate a Merkle tree, the root of which was written on chain and allow the user to securely generate a Merkle proof of discrete elements of that array (user authorizes Snap to decrypt, provides discrete field(s) to share, Snap generates and returns Merkle proofs to the connected dApp). 

Thanks in advance and special thanks to @Montoya for the Discord invite and link to this discussion forum!

**Comment by Montoya** [comment, marked as answer]
Hi Shawn! Thanks for reaching out. It turns out `eth_decrypt` isn't working because the snap needs the `eth_accounts` permission in order to use it. Please try adding that permission and see if it works. 

The current version of `eth-sig-util` may be incompatible with SES... @rekmarks will have to look into it to confirm.

**Reply by shawnmitchell** [reply]
```
  "initialPermissions": {
    "snap_confirm": {},
    "eth_accounts": {
      "requiredMethods": [
        "eth_decrypt"
      ]
    }
  },
```
Works like a charm! Thank you so much!

**Reply by shawnmitchell** [reply]
For posterity, Frederik determined that `eth-sig-util` _is_ supported at least as of `snaps-cli` version `0.10.5`
